evm_loader/program/src/entrypoint.rs

```diff
@@ -484,7 +485,11 @@ fn process_instruction<'a>(
             let account_info_iter = &mut accounts.iter();
             let storage_info = next_account_info(account_info_iter)?;
 
-            let mut storage = StorageAccount::restore(storage_info)?;
+            let mut storage = match StorageAccount::restore(storage_info) {
```

Comment:
You can use the `map_err` function here. It conveys more precisely what you want to do.
```
            let mut storage = StorageAccount::restore(storage_info).map_err(|err| {
                if err == ProgramError::InvalidAccountData {EVMLoaderError::StorageAccountUninitialized.into()}
                else {err}
            })?;
```